src/Fantomas/CodePrinter.fs

```diff
@@ -2616,7 +2616,11 @@ and genPat astContext pat =
     match pat with
     | PatOptionalVal(s) -> !- (sprintf "?%s" s)
     | PatAttrib(p, ats) -> genOnelinerAttributes astContext ats +> genPat astContext p
-    | PatOr(p1, p2) -> genPat astContext p1 +> sepNln -- "| " +> genPat astContext p2
+    | PatOr(p1, p2) ->
+        genPat astContext p1
+        +> enterNodeTokenByName pat.Range "BAR"
```

Comment:
See the previous remark, I'm guessing here you want to add `sepNlnConsideringTriviaContentBefore`.
then `enterNodeTokenByName` and then the pipe `-- "| "`

Reply:
yes, that did it!